src/webgpu/api/validation/attachment_compatibility.spec.ts

```diff
@@ -135,6 +137,22 @@ class F extends ValidationTest {
       sampleCount,
     });
   }
+
+  getTextureFormatsRequiredExtensions(
+    formats: (DepthStencilFormat | undefined)[]
```

Comment:
Not really, `undefined` is added on line 26. In addition, we can't use GPUTextureFormat because of the type check. Which is very strict and very silly. For example: 
very strict: type subset and type will be thought mismatch and not allowed. For example, DepthStencilFormat is a subset of GPUTextureFormat / AllTextureFormats, but it will report mismatch if the formats are DepthStencilFormat and you use kAllTextureFormatInfo(format)).
 very silly: you may exclude `undefined` somewhere, but it will complain type mismatch because it doesn't know you exclude `undefined` in its caller or caller's caller, etc. It seems to me that you need to exclued `undefined` exactly where you use it.

Reply:
Ah, sorry, didn't realize why undefined was needed, but it makes sense and should be kept.

Let me think about the other problem.

Reply:
I didn't have an issue with it. I will push a commit to your branch.